core/src/streamer.rs

```diff
@@ -27,21 +27,20 @@ fn recv_loop(
     channel_tag: &'static str,
 ) -> Result<()> {
     loop {
-        let msgs = SharedPackets::default();
+        let mut msgs = Packets::default();
         loop {
             // Check for exit signal, even if socket is busy
             // (for instance the leader trasaction socket)
             if exit.load(Ordering::Relaxed) {
                 return Ok(());
             }
-            if msgs.write().unwrap().recv_from(sock).is_ok() {
-                let len = msgs.read().unwrap().packets.len();
+            if let Ok(len) = msgs.recv_from(sock) {
                 submit(
```

Comment:
We should probably replace this metrics with ```inc_new_counter_info!()``` instead of ```submit()```. In past, receiver was being called from multiple (non test) points in code. We used ```channel_tag``` to track which channel was receiving how much data. Now only ```fetch_stage``` is calling receiver. So probably, we can either move this out to fetch stage, or remove the channel tag, and just track receiver_count.

Alternatively, if ```packets_recv_count``` is same as this data, we can remove it and use the other metrics.

Reply:
please have a look at the call stack and our current dashboards and make a determination for me ;)